docs/docs/plugin.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Extend Marathon with Plugins
+---
+
+
+# Extend Marathon with Plugins
+
+<div class="alert alert-danger" role="alert">
+  <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span> Available in Marathon Version 0.12+ <br/>
+  The Marathon plugin functionality is considered Alpha! <br/>
+  You can use this feature at you own risk. <br/>
+  We might want to add, change or delete any functionality described in this document. 
+</div>
+
+## Overview
+
+<p class="text-center">
+  <img src="{{ site.baseurl}}/img/plugin-mechanism.png" width="552" height="417" alt="">
+</p>
+
+
+Starting with Marathon 0.12 we provide the ability to extend functionality in Marathon with plugins.
+Every extension comprises this functionality:
+
+- __Extension Aware Functionality:__ Marathon makes specific functionality available for customization. There are hooks implemented into the main system, that can be refined and extended with external plugins.
+- __Extension Interface:__ The extended functionality is defined as scala trait (java interface). In order to extend this functionality, the plugin writer has to implement that interface.
+- __Plugin Extension:__ This is the implementation of a defined Extension Interface. The implementation needs to be compiled and packaged into one or more separate jar files. 
+- __Plugin Descriptor:__ The descriptor defines, which set of plugins should be enabled in one instance of Marathon.
+  
+
+
+## Plugin Interface
+
+A separate Marathon Plugin interface jar is published with every Marathon Release on that maven repository:
+
+```
+http://downloads.mesosphere.io/marathon
+```
+
+with following dependency:
+
+SBT:
+
+```
+"mesosphere.marathon" %% "plugin-interface" % "x.x.x" % "provided"
+
+```
+
+Maven:
+
+```
+<dependency>
+    <groupId>mesosphere.marathon</groupId>
+    <artifactId>plugin-interface_2.11</artifactId>
+    <version>x.x.x</version>
+</dependency>
+```
+
+Please make sure, that you always use the same version of the plugin interface as the Marathon version in production.
+
+
+
+## Extension Functionality
+
+To create an extension, you have to implement an extension interface from the plugin interface method, that can be loaded by the ServiceLoader.
```

Comment:
s/that can be loaded by the ServiceLoader./and put it in a location available to Marathon, so that it can be loaded by the ServiceLoader./